pkg/config/analysis/analyzers/envoyfilter/envoyfilter.go

```diff
@@ -45,20 +45,88 @@ func (*EnvoyPatchAnalyzer) Metadata() analysis.Metadata {
+			// indicating that the operation will be ignored when applyTo is set to ROUTE_CONFIGURATION,
+			// or HTTP_ROUTE
+			if patch.ApplyTo == network.EnvoyFilter_ROUTE_CONFIGURATION || patch.ApplyTo == network.EnvoyFilter_HTTP_ROUTE {
+				// provide an error message indicating a mismatch between the operation type and the filter type
+				message := msg.NewEnvoyFilterUsesAddOperationIncorrectly(r)
+
+				if line, ok := util.ErrorLine(r, fmt.Sprintf(util.EnvoyFilterConfigPath, index)); ok {
+					message.Line = line
+				}
+
+				c.Report(collections.IstioNetworkingV1Alpha3Envoyfilters.Name(), message)
+			}
+		} else if patch.Patch.Operation == network.EnvoyFilter_Patch_REMOVE {
+			// the REMOVE operation is ignored when applyTo is set to ROUTE_CONFIGURATION, or HTTP_ROUTE.
+			if patch.ApplyTo == network.EnvoyFilter_ROUTE_CONFIGURATION || patch.ApplyTo == network.EnvoyFilter_HTTP_ROUTE {
+				// provide an error message indicating a mismatch between the operation type and the filter type
+				message := msg.NewEnvoyFilterUsesRemoveOperationIncorrectly(r)
+
+				if line, ok := util.ErrorLine(r, fmt.Sprintf(util.EnvoyFilterConfigPath, index)); ok {
+					message.Line = line
+				}
+
+				c.Report(collections.IstioNetworkingV1Alpha3Envoyfilters.Name(), message)
+			} else if ef.Priority == 0 {
+				// Provide a warning indicating that no priority was set and a relative operation was used
+				message := msg.NewEnvoyFilterUsesRelativeOperation(r)
+
+				if line, ok := util.ErrorLine(r, fmt.Sprintf(util.EnvoyFilterConfigPath, index)); ok {
+					message.Line = line
+				}
+
+				c.Report(collections.IstioNetworkingV1Alpha3Envoyfilters.Name(), message)
+			}
+		} else if patch.Patch.Operation == network.EnvoyFilter_Patch_REPLACE {
+			// the REPLACE operation is only valid for HTTP_FILTER and NETWORK_FILTER.
+			if patch.ApplyTo != network.EnvoyFilter_NETWORK_FILTER && patch.ApplyTo != network.EnvoyFilter_HTTP_FILTER {
+				// provide an error message indicating an invalid filter type
+				message := msg.NewEnvoyFilterUsesReplaceOperationIncorrectly(r)
+
+				if line, ok := util.ErrorLine(r, fmt.Sprintf(util.EnvoyFilterConfigPath, index)); ok {
+					message.Line = line
+				}
+
+				c.Report(collections.IstioNetworkingV1Alpha3Envoyfilters.Name(), message)
+			} else {
+				count := 0
+				for _, name := range patchFilterNames {
+					if instanceName.String() == name {
+						count++
+						break
+					}
+				}
+				if count > 0 && ef.Priority == 0 {
+					// there is more than one envoy filter using the same name with the REPLACE operation
+					// provide a warning indicating that no priority was set and a relative operation was used
+					message := msg.NewEnvoyFilterUsesRelativeOperation(r)
+
+					if line, ok := util.ErrorLine(r, fmt.Sprintf(util.EnvoyFilterConfigPath, index)); ok {
+						message.Line = line
+					}
+					c.Report(collections.IstioNetworkingV1Alpha3Envoyfilters.Name(), message)
```

Comment:
These same lines are repeated in 4 places. Would be good to move to a separate function.